src/runtime/refiner.ts

```diff
@@ -526,6 +551,10 @@ export class FieldNamePrimitive extends RefinementExpression {
     return new Set<string>([this.value]);
   }
 
+  getQueryNames(): Set<string> {
+    return new Set<string>();
+  }
```

Comment:
provide default impl on superclass rather than repeating this in each subclass.

Reply:
Done.

P.S. We have a bunch of visitor patterns in our code base, it would be nice to work out a higher order interface that we could implement for them.

Then we could add `T implements Visitable<T>` to the type and something like
`children(): T[] { return [self.rightExpr self.rightExpr]}` in the nodes
Giving us

```
getQueryNames(): Set<string> {
    return this.visit(x => x.getQueryNames(), merge=Visitor.unionSets);
}

eval(): Set<string> {
    return this.visit(x => x.eval(), merge=this.applyOperator)
}
```

TLDR: Josh would like more Higher order types to avoid code duplication and implementation burden.